airbyte-cdk/python/airbyte_cdk/sources/declarative/low_code_component_schema.yaml

```diff
@@ -0,0 +1,988 @@
+"$schema": http://json-schema.org/draft-07/schema#
+"$id": https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/low_code_component_schema.yaml
+title: DeclarativeSource
+type: object
+description: An API source that extracts data according to its declarative components
+version: 1.0.0
+required:
+  - type
+  - check
+  - streams
+  - version
+properties:
+  type:
+    type: string
+    enum: [DeclarativeSource]
+  check:
+    "$ref": "#/definitions/CheckStream"
+  streams:
+    type: array
+    items:
+      "$ref": "#/definitions/DeclarativeStream"
+  version:
+    type: string
+  schemas:
+    "$ref": "#/definitions/Schemas"
+  spec:
+    "$ref": "#/definitions/Spec"
+definitions:
+  AddedFieldDefinition:
```

Comment:
should we / can we set additionalProperties to false for all components so developers get a compile-time error on unknown fields?

Reply:
At this moment we can't set this to false because we don't know what fields are on each model. So when we do the extra step of assigning options to component fields in the dereferenced manifest, we have to blanket apply them on every component. If we set this to false, we'll start getting validation errors.

There is a world where we get a little more targeted, but that would have to come after we've generated the models so we can infer the type hints to know what to add onto the object, and which to ignore. Built this way, we're not locked out of moving towards this approach later.

Reply:
that makes sense. thanks for confirming!